src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -820,6 +846,17 @@ private unsafe void DisposeInternal()
             });
             ForTestingPurposes?.DisposeLog?.Invoke(Name, "Disposed DocumentsStorage");
 
+            var clusterTransactionsTask = _clusterTransactionsTask;
+            if (clusterTransactionsTask != null)
+            {
+                ForTestingPurposes?.DisposeLog?.Invoke(Name, "Waiting for cluster transactions executor task to complete");
+                exceptionAggregator.Execute(() =>
+                {
+                    clusterTransactionsTask.Wait();
```

Comment:
do we want to set a time here, so we will not wait forever if something is hanging or taking too much time... (although for other places we don't set a time)

Reply:
We do wait for the indexes (and we don't set time there as well).